[new product] add Coder (#5496)
captn3m0/endoflife.date: products/coder.md

---
title: Coder
category: server-app
iconSlug: coder
permalink: /coder
versionCommand: coder version
releasePolicyLink: https://coder.com/docs/install/releases
changelogTemplate: https://github.com/coder/coder/releases/tag/v__LATEST__
releaseDateColumn: true

> eoasColumn: Active Support

Reviewer: ```suggestion
```

Already the default value.

Author: Without this, the column is hidden and does not show up.

Reviewer: Oups, you're right, sorry.

Author: I wanted to add this column. I will submit a follow up PR.

Reviewer: Done in #5590.

  latestReleaseDate: 2024-07-23

- releaseCycle: "2.12"
  releaseDate: 2024-06-04
  eoas: false # releaseDate 2.14
  eol: false # releaseDate 2.15
  latest: "2.12.5"
  latestReleaseDate: 2024-07-22

- releaseCycle: "2.11"
  releaseDate: 2024-05-07
  eoas: 2024-07-02 # releaseDate 2.13
  eol: 2024-08-06 # releaseDate 2.14
  latest: "2.11.4"
  latestReleaseDate: 2024-06-24

- releaseCycle: "2.10"
  releaseDate: 2024-04-03
  eoas: 2024-06-04 # releaseDate 2.12
  eol: 2024-07-02 # releaseDate 2.13
  latest: "2.10.3"
  latestReleaseDate: 2024-05-22

- releaseCycle: "2.9"
  releaseDate: 2024-03-07
  eoas: 2024-05-07 # releaseDate 2.11
  eol: 2024-06-04 # releaseDate 2.12
  latest: "2.9.4"
  latestReleaseDate: 2024-04-22

- releaseCycle: "2.8"
  releaseDate: 2024-02-06
  eoas: 2024-04-03 # releaseDate 2.10
  eol: 2024-05-07 # releaseDate 2.11
  latest: "2.8.5"
  latestReleaseDate: 2024-03-09

- releaseCycle: "2.7"
  releaseDate: 2024-01-01
  eoas: 2024-03-07 # releaseDate 2.9
  eol: 2024-04-03 # releaseDate 2.10
  latest: "2.7.3"
  latestReleaseDate: 2024-03-04

---

> [Coder](https://coder.com) is an open-source platform for creating and managing developer workspaces on your preferred

> > clouds and servers.

Reviewer: ```suggestion
> clouds and servers.

There is a new minor release of Coder on the first Tuesday of each month. Minor releases are
supported for three months with bug and security fixes the first month, major bug and security
fixes the second month and only security fixes the third month.
```

WDYT of this description ?

Author: Looks good for now and matches the release policy exactly. Thank you.

---
title: Coder
category: server-app

> tags: app

Reviewer: ```suggestion
```

Categories cannot be used a a tag.

> title: Coder

Reviewer: ```suggestion
---
title: Coder
```

category: server-app

> tags: remote cde terraform ide development docker kubernetes

Reviewer: ```suggestion
tags: build-tool
```

Only a limited set of tags is accepted, see https://endoflife.date/recommendations. The closest I could find is `build-tool` for now (we don't have an `ide` nor a `developer-tool` tag). WDYT ?

Author: I think. It doesn't come under a `build-tool` what do you think about supporting a new tag like `ide` or/and `development`?

Reviewer: Not for now, this should be done in a separate PR if there is enough candidate (we are not adding new tags unless it is the vendor name or applicable to at least three products (see https://endoflife.date/contribute, previously shared the wrong link).

So for now I will just delete the tags (`app` is a category and cannot be used as a tag).

Author: I see app being used as a tag in many apps. 
`server-app` is a category.

iconSlug: coder
permalink: /coder
versionCommand: coder version

> releasePolicyLink: https://coder.com/docs/install/releases#release-schedule

Reviewer: ```suggestion
releasePolicyLink: https://coder.com/docs/install/releases
```

All the page is interesting IMO.

changelogTemplate: https://github.com/coder/coder/releases/tag/v__LATEST__
releaseDateColumn: true
eoasColumn: Mainline
eoasWarnThreshold: 30
eolColumn: Stable
eolWarnThreshold: 60
eoesColumn: Security Support

> eoasWarnThreshold: 150

Reviewer: ```suggestion
eoesWarnThreshold: 150
```

Did you mean this ?.

Author: Yes. Thanks for catching the typo.

auto:
  methods:
  - git: https://github.com/coder/coder.git

releases:
- releaseCycle: "2.13"
  releaseDate: 2024-07-02
  eoas: 2024-08-06
  eol: 2024-09-03
  eoes: false

>   latest: "2.13.1"
>   latestReleaseDate: 2024-07-20

Reviewer: ```suggestion
  latest: "2.13.2"
  latestReleaseDate: 2024-07-23
```

Author: Thank you. I wanted to bump this version.

- releaseCycle: "2.12"
  releaseDate: 2024-06-04
  eoas: 2024-07-02
  eol: 2024-08-06
  eoes: false
  latest: "2.12.5"
  latestReleaseDate: 2024-07-22

- releaseCycle: "2.11"
  releaseDate: 2024-05-07
  eoas: 2024-06-04
  eol: 2024-07-02
  eoes: false

>   latest: "2.11.3"
>   latestReleaseDate: 2024-06-21

Reviewer: ```suggestion
  latest: "2.11.4"
  latestReleaseDate: 2024-06-24
```

Author: Thank you.

- releaseCycle: "2.10"
  releaseDate: 2024-04-03
  eoas: 2024-05-07
  eol: 2024-06-04
  eoes: false
  latest: "2.10.3"
  latestReleaseDate: 2024-05-22

- releaseCycle: "2.9"
  releaseDate: 2024-03-07
  eoas: 2024-04-03
  eol: 2024-05-07
  eoes: true
  latest: "2.9.4"
  latestReleaseDate: 2024-04-22

- releaseCycle: "2.8"
  releaseDate: 2024-02-06
  eoas: 2024-03-07
  eol: 2024-04-03
  eoes: true

>   latest: "2.8.4"
>   latestReleaseDate: 2024-03-09

Reviewer: ```suggestion
  latest: "2.8.5"
  latestReleaseDate: 2024-03-09
```

- releaseCycle: "2.7"
  releaseDate: 2024-01-01
  eoas: 2024-02-06
  eol: 2024-03-07
  eoes: true
  latest: "2.7.3"
  latestReleaseDate: 2024-03-04

---

> > [Coder](https://github.com/coder/coder) is an open-source cloud development environment (CDE) that you host in your cloud or on-premises. With Coder, you can deploy environments that provide the infrastructure, IDEs, and tools your developers need. Upgrade to Coder Enterprise to gain enhanced security, governance, and observability for your platform teams.

Reviewer: ```suggestion
> [Coder](https://coder.com) is an open-source platform for creating and managing developer workspaces on your preferred
> clouds and servers.
```

Shorter description from https://coder.com/docs.

> > [Coder](https://github.com/coder/coder) is an open-source cloud development environment (CDE) that you host in your cloud or on-premises. With Coder, you can deploy environments that provide the infrastructure, IDEs, and tools your developers need. Upgrade to Coder Enterprise to gain enhanced security, governance, and observability for your platform teams.

Reviewer: The product needs a description of the release and support policy. From what I understood:

- the last three versions are supported
- there are three stages of support:
  - mainline: bug and security fixes
  - stable: major bug and security fixes
  - security: only security fixes

Unfortunately endoflife.date structure is not suited for this kind of support. You used the `eoesColumn` for documenting security support but this does not fit : extended support should be reserved for support past EOL, such as commercial support. So given the security phase is a standard support phase, it should be documented in the `eolColumn`.

I propose to document both mainline and stable versions in the active support. I agree it's not ideal, but it's close to reality and best fit with endoflife.date way of documenting versions.

WDYT ?

Reviewer: And given what I understood of the support policy, we could also apply the following rule:

- eoas(x) = releaseDate(x+2)
- eol(x) = releaseDate(x+3)

Author: This sounds nice. I would appreciate some help on how can we apply these rules?

Author: > Unfortunately endoflife.date structure is not suited for this kind of support. You used the eoesColumn for documenting security support but this does not fit : extended support should be reserved for support past EOL, such as commercial support. So given the security phase is a standard support phase, it should be documented in the eolColumn.

I agree.

Author: > I propose to document both mainline and stable versions in the active support. I agree it's not ideal, but it's close to reality and best fit with endoflife.date way of documenting versions.

I agree and we do support both of them. Its just that once a release is cut it does not get new features but only bug and security fixes.

Author: I think I have fixed it for now. And I like the current preview. I hope it will be automated.

Reviewer: Unfortunately for now EOAS / EOL cannot be automated unless there is a page on the official website documenting the dates.
